{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Download dataset\n",
    "\n",
    "Download the dataset from [EasyGWAS](https://easygwas.ethz.ch/down/dataset/download/1/)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "from zipfile import ZipFile\n",
    "\n",
    "\n",
    "def extract():\n",
    "    with ZipFile(\"AtPolyDB_(call_method_75__Horton_et_al).zip\") as myzip:\n",
    "        myzip.extractall()\n",
    "extract()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.26 64-bit (2 Apr 2022)           www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2022 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to plink.log.\n",
      "Options in effect:\n",
      "  --file genotype\n",
      "\n",
      "7806 MB RAM detected; reserving 3903 MB for main workspace.\n",
      ".ped scan complete (for binary autoconversion).01111111111111111111111111112121212121212121212121212131313131313131313131313131414141414141414141414141414151515151515151515151515151616161616161616161616161617171717171717171717171717181818181818181818181818181919191919191919191919191920202020202020202020202020212121212121212121212121212222222222222222222222222223232323232323232323232323242424242424242424242424242525252525252525252525252526262626262626262626262626272727272727272727272727272828282828282828282828282828292929292929292929292929293030303030303030303030303031313131313131313131313131323232323232323232323232323333333333333333333333333334343434343434343434343434353535353535353535353535353636363636363636363636363637373737373737373737373737383838383838383838383838383939393939393939393939393940404040404040404040404040414141414141414141414141414242424242424242424242424242434343434343434343434343434444444444444444444444444445454545454545454545454545464646464646464646464646464747474747474747474747474748484848484848484848484848494949494949494949494949495050505050505050505050505051515151515151515151515151525252525252525252525252525353535353535353535353535354545454545454545454545454555555555555555555555555555656565656565656565656565657575757575757575757575757575858585858585858585858585859595959595959595959595959606060606060606060606060606161616161616161616161616162626262626262626262626262636363636363636363636363636464646464646464646464646465656565656565656565656565666666666666666666666666666767676767676767676767676768686868686868686868686868696969696969696969696969697070707070707070707070707071717171717171717171717171717272727272727272727272727273737373737373737373737373747474747474747474747474747575757575757575757575757576767676767676767676767676777777777777777777777777777878787878787878787878787879797979797979797979797979808080808080808080808080808181818181818181818181818182828282828282828282828282838383838383838383838383838484848484848484848484848485858585858585858585858585858686868686868686868686868687878787878787878787878787888888888888888888888888888989898989898989898989898990909090909090909090909090919191919191919191919191919292929292929292929292929293939393939393939393939393949494949494949494949494949595959595959595959595959596969696969696969696969696979797979797979797979797979898989898989898989898989899999999999999999999999999100%\n",
      "Performing single-pass .bed write (214051 variants, 1307 people).\n",
      "--file: plink.bed + plink.bim + plink.fam written.353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394%\n"
     ]
    }
   ],
   "source": [
    "# use plink extract\n",
    "!plink --file genotype"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Mapping files: 100%|██████████| 3/3 [00:00<00:00,  5.70it/s]\n"
     ]
    }
   ],
   "source": [
    "from pandas_plink import read_plink\n",
    "import pandas as pd\n",
    "\n",
    "\n",
    "bim, fam, bed = read_plink(\"plink\")\n",
    "genotypes = bed.compute()\n",
    "phenotypes = pd.read_csv(\"phenotypes.pheno\", sep=\" \")"
   ]
  }
 ],
 "metadata": {
  "interpreter": {
   "hash": "ed20a1528af572cc0c36245e4e9dadf4c663c0bea82a328dfd54063ee9b600dc"
  },
  "kernelspec": {
   "display_name": "Python 3.10.4 64-bit ('bio')",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.4"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
